power_grid_model_c/power_grid_model/include/power_grid_model/component/load_gen.hpp

```diff
@@ -17,6 +17,18 @@
 
 namespace power_grid_model {
 
+struct load_appliance_t {};
+struct gen_appliance_t {};
```

Comment:
There is ambiguity here between (a)symmetric calculation type and (a)symmetric component type. Idem for power sensor

To work around that, we may actually want to create different tags for component type and calculation type symmetry. My proposal is to do that in `base.hpp`

Alternatively, we may actually want to create 4 tags here:

```suggestion
struct sym_load_appliance_t {};
struct asym_load_appliance_t {};
struct sym_gen_appliance_t {};
struct asym_gen_appliance_t {};
```

that would fix the ambiguity between (a)symmetric calculation type and (a)symmetric component type

Reply:
The ambiguity should be solved by the name of template argument, not creating another tag type. E.g.

```c++
template<symmetry_tag loadgen_symmetry>
class LoadGen {

template<symmetry_tag calculation_symmetry>
LoadGenMathInput<calculation_symmetry> calc_params();
};

```